tiledb/sm/cpp_api/map.h

```diff
@@ -595,22 +595,31 @@ class Map {
    * @param ctx Context
    * @param uri URI of map.
    */
-  Map(const Context& ctx, const std::string& uri)
+  Map(const Context& ctx,
+      const std::string& uri,
+      const std::vector<std::string>& attributes = std::vector<std::string>())
       : schema_(MapSchema(ctx, (tiledb_kv_schema_t*)nullptr))
-      , uri_(uri)
-      , iter_(*this)
-      , iter_end_(*this, true) {
+      , uri_(uri) {
     tiledb_kv_t* kv;
     ctx.handle_error(tiledb_kv_alloc(ctx, uri.c_str(), &kv));
     kv_ = std::shared_ptr<tiledb_kv_t>(kv, deleter_);
-    ctx.handle_error(tiledb_kv_open(ctx, kv, nullptr, 0));
+
+    if (attributes.empty()) {
+      ctx.handle_error(tiledb_kv_open(ctx, kv, nullptr, 0));
+    } else {
+      auto attribute_num = (unsigned)attributes.size();
+      const char** c_attributes = new const char*[attribute_num];
+      for (unsigned i = 0; i < attribute_num; ++i)
+        c_attributes[i] = attributes[i].c_str();
+      ctx.handle_error(tiledb_kv_open(ctx, kv, c_attributes, attribute_num));
```

Comment:
I believe this will leak `c_attributes` on the error path (when an exception is thrown)